plugins/GCodeWriter/GCodeWriter.py

```diff
@@ -69,8 +70,13 @@ def _serialiseSettings(self, settings):
         prefix_length = len(prefix)
 
         global_stack = Application.getInstance().getGlobalContainerStack()
-        container_with_profile = global_stack.findContainer({"type": "quality"})
-        serialized = container_with_profile.serialize()
+        # Collecting all data from the top of the stack
+        temporarySettings= InstanceContainer("")
+        temporarySettings.setDefinition(ContainerRegistry.getInstance().findDefinitionContainers(id="fdmprinter")[0])
+        for key in global_stack.getAllKeys():
```

Comment:
This does not get all settings; This only gets settings for single extrusion machines.